plugin.video.earthcam/resources/lib/earthcam.py

```diff
@@ -123,52 +122,48 @@ def resolveURL(self, name, url):
                 self.addLink(label, liveurl, 9, infoLabels, infoArt, len(pageids))
                 # label      = '%s,%s - Live (FLV)'%(results["long_title"],results["country"])
                 # infoLabels = {"mediatype":"episode","label":label ,"title":label,"plot":plot}
                 # liveurl = ('%s%s'%(results["streamingdomain"],results["livestreamingpath"]))
                 # self.addLink(label, liveurl, 9, infoLabels, infoArt, len(pageids))
             elif  results["timelapseon"] == "true":
-                label      = '%s,%s -TimeLapse'%(results["long_title"],results["country"])
-                infoLabels = {"mediatype":"episode","label":label ,"title":label,"plot":plot}
+                label      = '%s - %s, %s Timelapse'%(results["camtext"], name,results["country"])
+                infoLabels = {"mediatype":"episode","label":label,"title":label,"plot":plot}
                 liveurl = ('http:%s%s'%(results["timelapsedomain"],results["timelapsepath"]))
                 self.addLink(label, liveurl, 9, infoLabels, infoArt, len(pageids))
             elif  results["archiveon"] == "true":
-                label      = '%s,%s - Archive'%(results["long_title"],results["country"])
-                infoLabels = {"mediatype":"episode","label":label ,"title":label,"plot":plot}
+                label      = '%s - %s, %s Archive'%(results["camtext"], name,results["country"])
+                infoLabels = {"mediatype":"episode","label":label,"title":label,"plot":plot}
                 liveurl = ('http:%s%s'%(results["archivedomain"],results["archivepath"]))
                 self.addLink(label, liveurl, 9, infoLabels, infoArt, len(pageids))
                 
 
     def prepareLink(self, url):
         log('prepareLink, url = ' + str(url))
-        if len(re.findall('http[s]?://www.youtube.com/watch', url)) > 0: return 'plugin://plugin.video.youtube/play/?video_id=%s'%(url.split('/watch?v=')[1])
-        elif url.lower().endswith(".m3u8"): url = url.replace('playlist', re.search(r'^([^#].+)\.m3u8$', self.openURL(url, True), re.MULTILINE).group(1))
+        try:
+            if len(re.findall('http[s]?://www.youtube.com/watch', url)) > 0: return 'plugin://plugin.video.youtube/play/?video_id=%s'%(url.split('/watch?v=')[1])
+            elif url.lower().endswith(".m3u8"): return url.replace('playlist', re.search(r'^([^#].+)\.m3u8$', self.openURL(url, True), re.MULTILINE).group(1))
+        except: return None
         return url
 
      
     def playVideo(self, name, url):
         log('playVideo')
-        liz  = xbmcgui.ListItem(name, path=self.prepareLink(url))
-        # if url.startswith('rtmp'):
-            # liz.setProperty('inputstreamaddon','inputstream.adaptive')
-            # liz.setProperty('inputstream.adaptive.manifest_type','hls')
+        url = self.prepareLink(url)
+        if url is None: return
+        liz = xbmcgui.ListItem(name, path=url)
+        if url.startswith('rtmp'):
```

Comment:
shouldn't it be `if *not* url.startswith('rtmp')`?

Reply:
Corrected typo.